pkg/util/conditions/condition_test.go

```diff
@@ -0,0 +1,265 @@
+package conditions
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	arofake "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned/fake"
+	"github.com/Azure/ARO-RP/pkg/util/cmp"
+)
+
+func TestSetCondition(t *testing.T) {
+	ctx := context.Background()
+	role := "master"
+	objectName := "cluster"
+	version := "unknown"
+
+	kubeclock = &clock.FakeClock{}
+	var transitionTime metav1.Time = metav1.Time{Time: kubeclock.Now()}
+
+	for _, tt := range []struct {
+		name      string
+		aroclient aroclient.Interface
+		input     operatorv1.OperatorCondition
+
+		expected arov1alpha1.ClusterStatus
+		wantErr  error
+	}{
+		{
+			name: "noop",
+			aroclient: arofake.NewSimpleClientset(&arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: objectName,
+				},
+			}),
+			expected: arov1alpha1.ClusterStatus{
+				OperatorVersion: version,
+				Conditions:      []operatorv1.OperatorCondition{},
+			},
+		},
+		{
+			name: "clean on version change",
+			aroclient: arofake.NewSimpleClientset(&arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: objectName,
+				},
+				Status: arov1alpha1.ClusterStatus{
+					Conditions: []operatorv1.OperatorCondition{
+						{
+							Type:   arov1alpha1.InternetReachableFromMaster.String(),
+							Status: operatorv1.ConditionFalse,
+						},
+					},
+					OperatorVersion: "?",
+				},
+			}),
+			expected: arov1alpha1.ClusterStatus{
+				Conditions: []operatorv1.OperatorCondition{
+					{
+						Type:   arov1alpha1.InternetReachableFromMaster.String(),
+						Status: operatorv1.ConditionFalse,
+					},
+				},
+				OperatorVersion: version,
+			},
+		},
```

Comment:
What are we testing in this case? That the version is being updated? If so, then probably we should reflect it in the case name

```suggestion
		{
			name: "update operator version",
			aroclient: arofake.NewSimpleClientset(&arov1alpha1.Cluster{
				ObjectMeta: metav1.ObjectMeta{
					Name: objectName,
				},
				Status: arov1alpha1.ClusterStatus{
					Conditions: []operatorv1.OperatorCondition{
						{
							Type:   arov1alpha1.InternetReachableFromMaster.String(),
							Status: operatorv1.ConditionFalse,
						},
					},
					OperatorVersion: "?",
				},
			}),
			expected: arov1alpha1.ClusterStatus{
				Conditions: []operatorv1.OperatorCondition{
					{
						Type:   arov1alpha1.InternetReachableFromMaster.String(),
						Status: operatorv1.ConditionFalse,
					},
				},
				OperatorVersion: version,
			},
		},
```